### PR TITLE
Feature/platfowner/new feature

### PR DESCRIPTION
**Pull request by platfowner**
Summary:
- Fix memory leak issue by changing deleteStateTreeVersion() logic
- Do some code refactoring

#171

### CODE REVIEW

File: db/state-util.js [hunk 1]
```diff
@@ -215,45 +213,30 @@ function deleteStateTree(node) {
...
+  node.resetProofHash();
+  numAffectedNodes++;
 
   for (const label of node.getChildLabels()) {
-    const childNode = node.getChild(label);
+    const child = node.getChild(label);
     node.deleteChild(label);
-    if (childNode.numParents() === 0) {
-      numAffectedNodes += deleteStateTreeVersion(childNode, version);
-    } else if (childNode.numParents() < 0) {
```
Comment (liayoo) [inline db/state-util.js:238]: question) do we not need to check/log this case anymore?

Comment (platfowner) [reply db/state-util.js:238]: We used to be using a separate count for this value so we were checking this. But now we're using a set and its 'size' for numParents() so it cannot be negative any more.

**Review by minsulee2 — APPROVED** [review]
LGTM

File: db/state-util.js (hunk 1 above)
Comment (liayoo) [reply db/state-util.js:238]: I see, thanks for the explanation!

**Review by liayoo — APPROVED** [review]
LGTM, thanks!

**Comment by platfowner** [comment]
Thanks for the fast review! Now merging..

**Pull request merged**